cpp/src/arrow/compute/api_vector.h

```diff
@@ -174,6 +181,23 @@ class ARROW_EXPORT SelectKOptions : public FunctionOptions {
   std::vector<SortKey> sort_keys;
 };
 
+/// \brief Rank options
+class ARROW_EXPORT RankOptions : public FunctionOptions {
+ public:
+  explicit RankOptions(SortOrder order = SortOrder::Ascending,
+                       NullPlacement null_placement = NullPlacement::AtEnd,
+                       Tiebreaker tiebreaker = Tiebreaker::First);
```

Comment:
Nit: `TieBreaker tie_breaker` to keep naming consistent.

Reply:
I'm keeping tiebreaker as one word elsewhere throughout so I think this is OK (?) but if not let me know and happy to change universally